crates/cairo-lint-core/src/fix.rs

```diff
@@ -206,6 +210,237 @@ impl Fixer {
         )
     }
 
+    /// Transforms nested `if-else` statements into a more compact `if-else if` format.
+    ///
+    /// Simplifies an expression by converting nested `if-else` structures into a single `if-else if`
+    /// statement while preserving the original formatting and indentation.
+    ///
+    /// # Arguments
+    ///
+    /// * `db` - Reference to the `SyntaxGroup` for syntax tree access.
+    /// * `node` - The `SyntaxNode` containing the expression.
+    ///
+    /// # Returns
+    ///
+    /// A `String` with the refactored `if-else` structure.
+    ///
+    
+    pub fn fix_collapsible_if_else(&self, db: &dyn SyntaxGroup, node: SyntaxNode) -> String {
+        // Call the transformation function to handle collapsible if-else
+        let fixed_text = self.transform_if_else(node.get_text(db));
+
+        fixed_text
+    }
+
+    // Transforms text to replace "else { if" pattern with "else if"
+    fn transform_if_else(&self, text: String) -> String {
```

Comment:
looks extremely complex, try working with the tree directly not the text. you can do it by replacing the `else` node with the `if` node text and prefix it with `else`

Reply:
got it, working on it!